astropy/io/fits/file.py

```diff
@@ -109,17 +124,16 @@ def __init__(self, fileobj=None, mode=None, memmap=None, overwrite=False,
         # Holds mmap instance for files that use mmap
         self._mmap = None
 
+        mode = _normalize_fits_mode(mode)
```

Comment:
I'm not sure this is entirely true: we do allow people to use things like 'rb' or 'wb' for the FITS mode itself right? So we need to be able to convert that to a 'normalized' FITS mode that is understood internally. Maybe this is just another wrong assumption I have about use cases but I thought that this behavior was supported even before this PR.

Reply:
No, the `mode` parameter for `fits.open` should be only in the keys of `IO_FITS_MODES` (`readonly` etc.). That was checked by the `if mode not in IO_FITS_MODES:` block previously (lines 121-122).

Reply:
Okay. In principle I don't see why this shouldn't be possible since there is already a well-defined mapping between conventional file modes and FITS modes, but I will change this now in order to get this PR approved.

Reply:
The mapping is not 1 to 1, `rb` can be used for readonly, copyonwrite, and denywrite. I think that the main reason for these specific names, and imo it would be confusing to support the usual file modes in addition. We can still discuss that in another issue if you want, but here it's better to focus on reading compressed files :wink: